Feature/54 Google Cloud inclusive properties-local.mk

anderskristian opened the pull request:
This PR contains

1. gcloud container clusters get-credentials before using kubectl.
2. introducing properties-local.mk
3. The properties-local.mk is "empty" with check that "project" and "region" are defined by the developer.
4. README.md shows sequence of make steps.

Commit:
added a properties-local.mk file to let user define own settings

Commit:
documented 'make' steps that is relevant

Commit:
google level make is now using 'properties-local.mk'

Commit:
Documentation of make "steps"

Commit:
one get-credentials too much

upgradingdave commented:
Thanks for this @anderskristian! 

I'll review this as soon as I have some time (hopefully later this week) 💪

upgradingdave committed:
Updated documentation and refactored directory structure

upgradingdave commented:
@falko - Can you review this when you have a chance? 

This moves custom configurations (such as custom region, machine type, cluster name, dns name, etc) into `.mk` files inside a `config` directory.

Please let me know what you think? If you give the 👍 , then I'll refactor azure and aws to match.

upgradingdave committed:
add `image-pull-secret` target to `camunda.mk`

This is required to install web modeler

upgradingdave committed:
Remove property files from version control

upgradingdave committed:
Add connectors-password target

upgradingdave committed:
Enable web modeler by default

upgradingdave committed:
ignore google config files

upgradingdave committed:
Remove extra unecessary web modeler ingress

upgradingdave committed:
Add hello webhook bpmn example

upgradingdave committed:
Add `aws/config` directory

upgradingdave committed:
Ignore properties files in aws directory

upgradingdave commented:
Refactored `aws` tls profile to use config directory. 

Still need to refactore other aws profiles as well as azure profiles.

Commit:
establish a .gitignore for Google.
It makes customer merge conflicts less.
It makes modularity / refactoring "easier"

Commit:
Improved Google sub menu

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda-community-hub/camunda-8-helm-profiles?pullRequest=55) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/camunda-community-hub/camunda-8-helm-profiles?pullRequest=55) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: upgradingdave<br/>:x: Anders K. Andersen<br/><hr/>**Anders K. Andersen** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda-community-hub/camunda-8-helm-profiles?pullRequest=55) it.</sub>

falko commented on ingress-nginx/camunda-values.yaml:
```diff
@@ -47,6 +49,24 @@ connectors:
 #      value: demo
 
+webModeler:
+  enabled: true
```
It's great to have the ingress configuration for the Web Modeler but I'd disable it by default.

falko commented on bpmn/hello_webhook.bpmn:
Why is this file required for this PR?

falko commented:
Hi @anderskristian,

sorry for not getting back to you earlier. I really like your proposal and recently came to the same conclusion that the configs should better be extracted from the `Makefile` in this new profile: https://github.com/camunda-community-hub/camunda-8-helm-profiles/tree/main/multi-region/dual-region

Would you still be interested to contribute? If so we would need you to:

1. Sign the Contributor License Agreement
2. Respond to my review questions ;-)
3. Rebase the PR on top of the latest `main` branch if possible

Cheers,
Falko

anderskristian commented:
I will still like to contribute.

1) I have signed the Contributor License Agreement
2) This thing is so long ago, that I have lost the touch of what's going on.
     I will accept your changes. 
3) I basically think we should redo the things from the current latest main.
     As I understand Camunda has simplified the deployment of Camunda 8.
     Therefore I cannot pin-point what to do 🧐
